src/superstring/core.clj

```diff
@@ -29,52 +32,52 @@
   "Create an alias for all public vars in ns in this ns."
   [namespace]
   `(do ~@(map
-          (fn [n] `(defalias ~(.sym n) ~(symbol (str (.ns n)) (str (.sym n)))))
+          (fn [^clojure.lang.Var n] `(defalias ~(.sym n) ~(symbol (str (.ns n)) (str (.sym n)))))
           (vals (ns-publics namespace)))))
 
 (alias-ns clojure.string)
-(def substring clojure.core/subs)
+(def ^{:tag String} substring clojure.core/subs)
 
-(defn ^Long length
+(defn length
   "Return the length of s."
-  [^String s]
-  {:pre [(string? s)]
-   :post [(integer? %)]}
-  (.length s))
+  (^long [^String s]
```

Comment:
That's the way I understand that you should do type hints in Clojure in order to be correct. Type hints at the top go on the var, which might not do what you want....